### PR TITLE
Fix master next build

### PR DESCRIPTION
**Pull request by Teemperor**

### CODE REVIEW

File: lldb/source/DataFormatters/FormatManager.cpp [hunk 1]
```diff
@@ -731,7 +731,7 @@ FormatManager::FormatManager()
   EnableCategory(m_vectortypes_category_name, TypeCategoryMap::Last,
                  lldb::eLanguageTypeObjC_plus_plus);
   EnableCategory(m_runtime_synths_category_name, TypeCategoryMap::Last,
-                 {lldb::eLanguageTypeObjC_plus_plus, lldb::eLanguageTypeSwift});
+                 lldb::eLanguageTypeSwift);
```
Comment (dcci) [inline lldb/source/DataFormatters/FormatManager.cpp:736]: Snap. Are you sure this is correct?

File: lldb/source/Core/ValueObjectChild.cpp [hunk 2]
```diff
@@ -168,19 +169,25 @@ bool ValueObjectChild::UpdateValue() {
...
               m_value.SetValueType(Value::eValueTypeFileAddress);
           } break;
-          case eAddressTypeLoad:
+          case eAddressTypeLoad: {
             // BEGIN SWIFT MOD
             // We need to detect when we cross TypeSystem boundaries,
             // e.g. when we try to print Obj-C fields of a Swift object.
-            if (parent->GetCompilerType().GetTypeSystem()->getKind() ==
-                GetCompilerType().GetTypeSystem()->getKind())
+            bool same_type_system = llvm::isa<ClangASTContext>(parent->GetCompilerType().GetTypeSystem()) &&
```
Comment (dcci) [inline lldb/source/Core/ValueObjectChild.cpp:176]: When did this change/break?

Comment (Teemperor) [reply lldb/source/Core/ValueObjectChild.cpp:176]: When I removed the Kind enum from TypeSystem (see the commit for the change).

File: lldb/source/DataFormatters/FormatManager.cpp (hunk 1 above)
Comment (Teemperor) [reply lldb/source/DataFormatters/FormatManager.cpp:736]: No, that's why I asked for review :)

Comment (dcci) [reply lldb/source/DataFormatters/FormatManager.cpp:736]: I think this is wrong, but to be sure, try the same patch on a stable branch and see if it passes. If not, then we probably need to reintroduce the API that takes a vector upstream.

File: lldb/source/Core/ValueObjectChild.cpp (hunk 2 above)
Comment (dcci) [reply lldb/source/Core/ValueObjectChild.cpp:176]: What was the rationale for that? The new way of writing this code looks clumsy and error prone.

Comment (Teemperor) [reply lldb/source/Core/ValueObjectChild.cpp:176]: That we don't have a centralized enum that has to list all TypeSystems inside LLDB (which would need to list all Mock and testing TypeSystems).

Anyway, the actual change that broke this is that the new RTTI implementation details are private (the ID member) so we can't just hijack that for this check. I could also make the RTTI backend public again but that would be a downstream patch. Or I upstream some kind of check that allows to check if two TypeSystems are of the same class.

File: lldb/source/DataFormatters/FormatManager.cpp (hunk 1 above)
Comment (Teemperor) [reply lldb/source/DataFormatters/FormatManager.cpp:736]: Seems like calling it twice (one for ObjC and once for Swift) is identical to the vector version. That seems less invasive.

**Pull request closed without merging**